src/controllers/universal-router/index.tsx

```diff
@@ -88,7 +88,7 @@ export class UniversalRouter extends Component<UniversalRouterProps> {
     return (
       <UniversalRouterContainer
         routes={routes}
-        history={getInferredHistory(history, location)}
```

Comment:
This was creating a new history on every re-render. Better store it in the class so we avoid such risk